### PR TITLE
Ovirt auth/vms/disks modules

### PR DESCRIPTION
**Pull request by machacekondra**
##### ISSUE TYPE

- New Module Pull Request
##### COMPONENT NAME

cloud/ovirt/ovirt_disks.py
cloud/ovirt/ovirt_vms.py
cloud/ovirt/ovirt_auth.py
##### ANSIBLE VERSION

```
2.2.0
```
##### SUMMARY

This patch consist of three basic modules for [oVirt](http://ovirt.org/).
1. _auth_ module, which manages authentication for oVirt.
2. _vms_ module, which manages Virtual Machines in oVirt.
3. _disks_ module, which manages Virtual Machines and floating disks in oVirt.

Module in _./cloud/misc/ovirt.py_ works with v3 REST API, this module works with v4 REST API.

### CODE REVIEW

**Comment by machacekondra** [comment]
CC @ryansb @detiber

**Comment by machacekondra** [comment]
CI fais, because this patchs depends on: https://github.com/ansible/ansible/pull/17364

**Comment by gregdek** [comment]
Thanks @machacekondra for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by machacekondra** [comment]
@mwperina @lukas-bednar

File: cloud/ovirt/ovirt_vms.py
Comment (detiber) [inline cloud/ovirt/ovirt_vms.py]: This parameter seems a bit awkward, almost seems like this should be a state.

File: cloud/ovirt/ovirt_vms.py
Comment (detiber) [inline cloud/ovirt/ovirt_vms.py:151]: Is this option limited to just pre-created disks? Being able to define a very simple disk here would be helpful for simple use cases.

File: cloud/ovirt/ovirt_vms.py
Comment (machacekondra) [reply cloud/ovirt/ovirt_vms.py]: OK, I will change it, thanks.

File: cloud/ovirt/ovirt_vms.py
Comment (machacekondra) [reply cloud/ovirt/ovirt_vms.py:151]: Yes currently it supports only pre-cretead disks. Ok, I will add some limited functionality for it.

File: cloud/ovirt/ovirt_auth.py
Comment (mykaul) [inline cloud/ovirt/ovirt_auth.py:25]: Surely we can print something - 'please ensure oVirt Python SDK 4 is installed' or something.

File: cloud/ovirt/ovirt_auth.py
Comment (mykaul) [inline cloud/ovirt/ovirt_auth.py]: authenticate -> authenticates. create -> creates

File: cloud/ovirt/ovirt_auth.py
Comment (mykaul) [inline cloud/ovirt/ovirt_auth.py]: return an ansible fact -> returns an Ansible fact

File: cloud/ovirt/ovirt_auth.py
Comment (mykaul) [inline cloud/ovirt/ovirt_auth.py]: token -> a token

File: cloud/ovirt/ovirt_auth.py
Comment (mykaul) [inline cloud/ovirt/ovirt_auth.py]: something like -> for example:

File: cloud/ovirt/ovirt_auth.py
Comment (mykaul) [inline cloud/ovirt/ovirt_auth.py]: usually something like -> for example:

File: cloud/ovirt/ovirt_auth.py
Comment (mykaul) [inline cloud/ovirt/ovirt_auth.py:59]: Shouldn't the default be secured? (thus True)

File: cloud/ovirt/ovirt_auth.py
Comment (mykaul) [inline cloud/ovirt/ovirt_auth.py]: for ever -> forever

File: cloud/ovirt/ovirt_auth.py
Comment (machacekondra) [reply cloud/ovirt/ovirt_auth.py:59]: Default is false, so by default we check system certificate store of the target machine if ca_file isn't passed.
If user pass 'insecure=True', we don't check certificate.

File: cloud/ovirt/ovirt_auth.py
Comment (mykaul) [inline cloud/ovirt/ovirt_auth.py]: generate -> generated

File: cloud/ovirt/ovirt_auth.py
Comment (mykaul) [inline cloud/ovirt/ovirt_auth.py:135]: sample or example? (here and below)

File: cloud/ovirt/ovirt_auth.py
Comment (mykaul) [inline cloud/ovirt/ovirt_auth.py]: URL of the oVirt engine API endpoint

File: cloud/ovirt/ovirt_auth.py
Comment (mykaul) [inline cloud/ovirt/ovirt_auth.py]: SSL -> SSL/TLS

File: cloud/ovirt/ovirt_auth.py
Comment (mykaul) [inline cloud/ovirt/ovirt_auth.py]: indication -> indicating

File: cloud/ovirt/ovirt_auth.py
Comment (mykaul) [inline cloud/ovirt/ovirt_auth.py]: Number of seconds to wait for response

File: cloud/ovirt/ovirt_auth.py
Comment (mykaul) [inline cloud/ovirt/ovirt_auth.py]: Shouldn't we compress by default?

File: cloud/ovirt/ovirt_disks.py
Comment (mykaul) [inline cloud/ovirt/ovirt_disks.py]: network -> storage

File: cloud/ovirt/ovirt_disks.py
Comment (mykaul) [inline cloud/ovirt/ovirt_disks.py]: This is for CHAP discovery or login?
(same for password)

File: cloud/ovirt/ovirt_disks.py
Comment (mykaul) [inline cloud/ovirt/ovirt_disks.py:117]: I expect in detach to have a different state?

File: cloud/ovirt/ovirt_disks.py
Comment (mykaul) [inline cloud/ovirt/ovirt_disks.py]: ID of the managed disk

File: cloud/ovirt/ovirt_vms.py
Comment (mykaul) [inline cloud/ovirt/ovirt_vms.py]: Is pause a separate state?

File: cloud/ovirt/ovirt_vms.py
Comment (mykaul) [inline cloud/ovirt/ovirt_vms.py]: don't -> doesn't
exists -> exist

File: cloud/ovirt/ovirt_vms.py
Comment (mykaul) [inline cloud/ovirt/ovirt_vms.py]: What is IEC 60027-2? Simply state if it's in MB/GB/TB or MiB/GiB/TiB (I hope the former).

File: cloud/ovirt/ovirt_vms.py
Comment (mykaul) [inline cloud/ovirt/ovirt_vms.py:102]: Is this affinity or to run on that host? I assume it's affinity.

File: cloud/ovirt/ovirt_vms.py
Comment (mykaul) [inline cloud/ovirt/ovirt_vms.py:156]: what about virtio-scsi?

File: cloud/ovirt/ovirt_vms.py
Comment (mykaul) [inline cloud/ovirt/ovirt_vms.py]: Why do we need to reboot it if we can hot-add memory, CPU, NICs and disks?!

File: cloud/ovirt/ovirt_auth.py
Comment (machacekondra) [reply cloud/ovirt/ovirt_auth.py:25]: It's actually printed [here](https://github.com/ansible/ansible/pull/17364/commits/0bafd8d70b779e93c0006a66a7e293d10f3a8003#diff-964a81589429c39efce1a9607db47ee6R44)

File: cloud/ovirt/ovirt_auth.py
Comment (machacekondra) [reply cloud/ovirt/ovirt_auth.py:135]: It's sample. It's parsed by the ansible documentation script.

File: cloud/ovirt/ovirt_auth.py
Comment (machacekondra) [reply cloud/ovirt/ovirt_auth.py]: Agree.

File: cloud/ovirt/ovirt_auth.py
Comment (mwperina) [inline cloud/ovirt/ovirt_auth.py:38]: I'd remove comma:

Every module can use this fact as C(auth) parameter to perform authentication.

File: cloud/ovirt/ovirt_auth.py
Comment (mwperina) [inline cloud/ovirt/ovirt_auth.py:73]: The maximum total time to wait for the response, in seconds.

->

The maximum total time in seconds to wait for the response.

File: cloud/ovirt/ovirt_auth.py
Comment (mwperina) [inline cloud/ovirt/ovirt_auth.py:127]: I'd remove the comma and rephrase:

Authentication facts required to perform authentication to oVirt.

File: cloud/ovirt/ovirt_auth.py
Comment (mwperina) [inline cloud/ovirt/ovirt_auth.py]: which store your

->

which stores your

File: cloud/ovirt/ovirt_auth.py
Comment (mwperina) [inline cloud/ovirt/ovirt_auth.py:98]:
```
# Create a vault with `ovirt_password` variable which store your
# oVirt user's password, and include that yaml file with variable
```

->

```
# Create a vault named `ovirt_password.yml`, which stores your
# oVirt user's password within `password` variable, and include that yaml file:
```

File: cloud/ovirt/ovirt_vms.py
Comment (mwperina) [inline cloud/ovirt/ovirt_vms.py:38]: Virtual Machine(VM)

->

Virtual Machine (VM)

File: cloud/ovirt/ovirt_vms.py
Comment (mwperina) [inline cloud/ovirt/ovirt_vms.py:39]: Since VM can hold many states in oVirt, this see notes to see how the states of the VM are handled.

->

There are a lot of possible VM states, so please take a look below at section describing how VM states are handled.

**Comment by gregdek** [comment]
@machacekondra A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by machacekondra** [comment]
ready_for_review once this PR: https://github.com/ansible/ansible/pull/17364 is merged

**Comment by gregdek** [comment]
Thanks @machacekondra for this new module. When this module receives 'shipit' comments from two community members and any 'needs_revision' comments have been resolved, we will mark for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

**Review by mwperina — APPROVED** [review]
Looks good to me, thanks

File: cloud/ovirt/ovirt_auth.py
Comment (ryansb) [inline cloud/ovirt/ovirt_auth.py:122]: This needs to have quotes around `{{ ovirt_auth }}` like the one on line 114

**Review by ryansb — COMMENTED** [review]
Overall these look good, just the one doc tweak. Please put that in a followup PR to these modules.

**Pull request merged**
